### DISCUSSION TITLE
Flags for things/people, and their names

### DISCUSSION THREAD

**Discussion by mefistotelis** [swfans/syndwarsfx#236] [General]
We have a thread about flags for Commands, so it makes sense to discuss things as well.

There are some hints in the debug HUD, but that's often not enough for me to decode the names:

```
        if (p_track_thing->Flag & TngF_Unkn00040000)
            draw_text(scr_x + 0, scr_y + ln*4, "Da", colour_lookup[ColLU_WHITE]);
        if (p_track_thing->Flag & TngF_WepRecoil)
            draw_text(scr_x + 20, scr_y + ln*4, "Re", colour_lookup[ColLU_WHITE]);
        if (p_track_thing->Flag & TngF_Unkn00020000)
            draw_text(scr_x + 40, scr_y + ln*4, "Si", colour_lookup[ColLU_WHITE]);
        if (p_track_thing->Flag & TngF_Destroyed)
            draw_text(scr_x + 60, scr_y + ln*4, "De", colour_lookup[ColLU_WHITE]);
        if (p_track_thing->Flag & TngF_Unkn0400)
            draw_text(scr_x + 80, scr_y + ln*4, "Ch", colour_lookup[ColLU_WHITE]);
        if (p_track_thing->Flag & TngF_Unkn0040)
            draw_text(scr_x + 100, scr_y + ln*4, "CI", colour_lookup[ColLU_WHITE]);
        if (p_track_thing->Flag & TngF_Unkn20000000)
            draw_text(scr_x + 120, scr_y + ln*4, "SAP", colour_lookup[ColLU_WHITE]);
        if (p_track_thing->Flag & TngF_StationrSht)
            draw_text(scr_x + 160, scr_y + ln*4, "Sta", colour_lookup[ColLU_RED]);
        if (p_track_thing->Flag & TngF_Unkn0800)
            draw_text(scr_x + 230, scr_y + ln*4, "TRIG", colour_lookup[ColLU_WHITE]);
```

**Comment by Moburma** [comment]
I'll have a look. I guess stating the obvious:

>         if (p_track_thing->Flag & TngF_WepRecoil)
>             draw_text(scr_x + 20, scr_y + ln*4, "Re", colour_lookup[ColLU_WHITE]);

Presumably that's recoil/reloading?

>         if (p_track_thing->Flag & TngF_Destroyed)
>             draw_text(scr_x + 60, scr_y + ln*4, "De", colour_lookup[ColLU_WHITE]);

Must be destroyed/dead

>         if (p_track_thing->Flag & TngF_StationrSht)
>             draw_text(scr_x + 160, scr_y + ln*4, "Sta", colour_lookup[ColLU_RED]);

Waiting at station?

**Reply by mefistotelis** [reply]
>  I guess stating the obvious: TngF_WepRecoil

Yes, I didn't removed the ones I did named..

> Waiting at station?

Some debug messages in 1st demo suggests it's "stationary shot", hence my `TngF_StationrSht`.
But I have no idea what the stationary shot is.

**Reply by Moburma** [reply]
> debug messages in 1st demo suggests it's "stationary shot"

Ahh I see, in that case that probably is it - i.e. it's the NPC standing still to take a shot (to "aim" first - is there even any accuracy penalty for moving and firing in the game)?

EDIT: Presumably to make e.g. LR Rifle attacks look more realistic and meaningful so they don't just run around blasting with it? I should probably actually test some of these!

**Comment by mefistotelis** [comment]
>  i.e. it's the NPC standing still to take a shot (to "aim" first - is there even any accuracy penalty for moving and firing in the game)?

I don't think so, but I'm not sure.

More rewrite is probably needed to fully understand the stationary shot. The flag seem to be set set at some point for every weapon besides launcher. Maybe it has some relation to overcharging. Not sure.

**Comment by Moburma** [comment]
Some trial and error I have some thoughts on these:

Ch - fairly sure this is Charging weapon. I was playing 2, 103, and the soldiers near the start were charging their lasers shooting at me when I killed them, and ended up dead with this status on. 

CI - This is something like Commands Idle or Commands Inactive. Seems to indicate a Peep who has no commands at all (so _not_ just they are finished). An example of this is most levels of course give no commands to the player and on these the player characters all appear as CI throughout. However, 2,103 starts with the player characters boarding a car using exec com, so these do not have that status at any point. Same with any NPCs in a level, with they are just comhead 0 they get this status.

TRIG - In combat, literally pulling the trigger fighting.

SAP - Not sure. 2,103 the soldiers guarding the transport tubes in the middle all have this status. I wonder if it's just they are doing support activities (laying wire and cerberus IFFs) and this just shows they are fulfilling a [Sapper](https://en.wikipedia.org/wiki/Sapping) role, but it does look like an acronym...

**Reply by Moburma** [reply]
Pretty much confirmed what I said above, SAP really just does seem to mean the unit was taking part in sapper activities. Another good example of this is either of the end levels, all the invisible support units have this status from laying triggerwire etc.

Si is a harder one. It seems to be some kind of support status used when other units gather behind a main unit that is waiting to storm and attack the player. Not sure what it stood for.
